calico/felix/frules.py

```diff
@@ -538,7 +538,7 @@ def update_ipsets(type,
     """
     for rule in rule_list:
         if rule.get('cidr') is None:
-            log.error("Invalid %s rule without cidr for %s : %s",
+            log.error("Invalid %s rule without cidr for %s : %s" %

```

Comment:
PEP8 isn't the bit that matters actually.

The reason you care is that using the latter will do the string interpolation every time the line is hit, regardless of whether the log will actually get emitted or not. In this case it doesn't matter much (error level logs will almost certainly get emitted, and it's not too expensive to do this particular reporting), but in some situations it can be extremely expensive.

For that reason, it's a slight performance enhancement to use the first case over the second.